File: BibliotecaUFABC/app/src/main/java/com/nintersoft/bibliotecaufabc/book_search_model/BookSearchDAO.java
package com.nintersoft.bibliotecaufabc.book_search_model;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

@Dao
public interface BookSearchDAO {

    @Query("SELECT * FROM search_items")
    List<BookSearchProperties> getAll();

    @Insert
    long insert(BookSearchProperties bookRenewalProperties);

    @Query("DELETE FROM search_items")
    void removeAll();
}

File: BibliotecaUFABC/app/src/main/java/com/nintersoft/bibliotecaufabc/model/search/BookSearchDAO.kt
package com.nintersoft.bibliotecaufabc.model.search

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

@Dao
interface BookSearchDAO {

    @Query("SELECT * FROM ${BookSearchContract.TABLE_NAME}")
    fun getAll() : LiveData<List<BookSearch>>

    @Insert
    fun insert(book : BookSearch) : Long

    @Delete
    fun remove(book : BookSearch) : Int

    @Query("DELETE FROM ${BookSearchContract.TABLE_NAME}")
    fun removeAll()
}